plugins/modules/nd_interface_flow_rules.py

```diff
@@ -0,0 +1,440 @@
+                            nodes_to_update.append({"flowNodeUuid": node_uuid, "operation": "MODIFY", "portsList": [{"port": port, "operation": "ADD"}]})
+                            nodes_to_add.append(dict(sanitize_dict(existing_node, ["flowNodeUuid", "portsList"]), **{"portsList": [{"port": port}]}))
+                        elif port not in ports_input and port in existing_port_set:
+                            port_uuid = next((existing_port["flowPortUuid"] for existing_port in existing_ports if existing_port["port"] == port))
+                            nodes_to_update.append(
+                                {
+                                    "flowNodeUuid": node_uuid,
+                                    "operation": "MODIFY",
+                                    "portsList": [{"flowPortUuid": port_uuid, "operation": "DELETE"}],
+                                }
+                            )
+                        else:
+                            nodes_to_add.append(sanitize_dict(existing_node, ["flowNodeUuid", "flowPortUuid"]))
+
+            else:
+                nodes_to_add.append(
+                    sanitize_dict(
+                        next((existing_node for existing_node in existing_nodes if existing_node["nodeId"] == node_id)),
+                        ["flowNodeUuid", "flowPortUuid"],
+                    )
+                )
+    return nodes_to_add, nodes_to_update
+
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(
+        insights_group=dict(type="str", aliases=["fab_name", "ig_name"]),
+        site_name=dict(type="str", aliases=["site"]),
+        flow_rule=dict(type="str", aliases=["interface_flow_rule", "flow_rule_name", "name"]),  # Not required to query all objects
+        flow_rule_status=dict(type="str", default="enabled", choices=["enabled", "disabled"], aliases=["status"]),
+        flow_rule_type=dict(type="str", choices=["port_channel", "physical", "l3out_sub_interface", "l3out_svi"], aliases=["type"]),
+        nodes=dict(
+            type="list",
+            elements="dict",
+            options=dict(
+                node_id=dict(type="str"),
+                node_name=dict(type="str"),
+                tenant=dict(type="str"),
+                l3out=dict(type="str"),
+                encap=dict(type="str"),
+                ports=dict(type="list", elements="str"),
+            ),
+        ),
+        subnets=dict(type="list", elements="str"),
+        state=dict(type="str", default="present", choices=["present", "absent", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=False,
+        required_if=[
+            ["state", "present", ["insights_group", "site_name", "flow_rule"]],
+            ["state", "absent", ["insights_group", "site_name", "flow_rule"]],
+            ["state", "query", ["insights_group", "site_name"]],
```

Comment:
why not make insights_group and site_name required?

Reply:
made the change.